tests/phpunit/tests/test-slugs.php

```diff
@@ -12,20 +19,122 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
 		self::create_language( 'fr_FR' );
 	}
 
-	public function test_term_slugs() {
-		$links_model = self::$model->get_links_model();
-		$pll_admin = new PLL_Admin( $links_model );
-		$pll_admin->term = new PLL_CRUD_Terms( $pll_admin );
-		$pll_admin->filters_term = new PLL_Admin_Filters_Term( $pll_admin ); // activate our filters
+	public function set_up() {
+		parent::set_up();
+
+		$options                       = PLL_Install::get_default_options();
+		$options['default_lang']       = 'en'; // Default language is the first one created, see self::wpSetUpBeforeClass().
+		$model                         = new PLL_Admin_Model( $options );
+		$links_model                   = new PLL_Links_Default( $model );
+		$this->pll_admin               = new PLL_Admin( $links_model );
+		$this->pll_admin->term         = new PLL_CRUD_Terms( $this->pll_admin ); // Activates our generic term filters.
+		$this->pll_admin->filters_term = new PLL_Admin_Filters_Term( $this->pll_admin );  // Activates our filters for admin.
+	}
 
+	public function test_term_slugs() {
 		$term_id = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'test' ) );
-		self::$model->term->set_language( $term_id, 'en' );
+		$this->pll_admin->model->term->set_language( $term_id, 'en' );
 
 		$_POST['term_lang_choice'] = 'fr';
-		$term_id = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'test' ) );
-		self::$model->term->set_language( $term_id, 'fr' );
+		$term_id                   = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'test' ) );
+		$this->pll_admin->model->term->set_language( $term_id, 'fr' );
 
 		$term = get_term( $term_id, 'category' );
 		$this->assertEquals( 'test-fr', $term->slug );
 	}
+
+	public function test_term_with_parents_sharing_same_name() {
+		$en_parent = self::factory()->term->create_and_get( array( 'taxonomy' => 'category', 'name' => 'test' ) );
+		$this->pll_admin->model->term->set_language( $en_parent->term_id, 'en' );
+
+		$this->assertInstanceOf( WP_Term::class, $en_parent );
+		$this->assertEquals( 'test', $en_parent->slug );
+
+		$_POST['term_lang_choice'] = 'en';
+		$_POST['parent']           = $en_parent->term_id;
+		$en                        = self::factory()->term->create_and_get( array( 'taxonomy' => 'category', 'name' => 'test', 'parent' => $en_parent->term_id ) );
+		$this->pll_admin->model->term->set_language( $en, 'en' );
+
+		$this->assertInstanceOf( WP_Term::class, $en );
+		$this->assertEquals( 'test-test', $en->slug );
+	}
+
+	public function test_translated_terms_with_parents_sharing_same_name() {
+		$en_parent = self::factory()->term->create_and_get( array( 'taxonomy' => 'category', 'name' => 'test' ) );
+		$this->pll_admin->model->term->set_language( $en_parent->term_id, 'en' );
+
+		$this->assertInstanceOf( WP_Term::class, $en_parent );
+		$this->assertEquals( 'test', $en_parent->slug );
```

Comment:
```suggestion
		$this->assertSame( 'test', $en_parent->slug );
```

![image](https://user-images.githubusercontent.com/1878479/195564808-f955ccb5-bbc0-4903-b559-95842102b757.png)

Reply:
Ouch! Copy/paste hell!

Reply:
Done in a0999f0